textdb/textdb-storage/src/main/java/edu/uci/ics/textdb/storage/reader/DataReader.java

```diff
@@ -72,25 +72,28 @@ public void open() throws DataFlowException {
             TopDocs topDocs = luceneIndexSearcher.search(dataReaderPredicate.getLuceneQuery(), Integer.MAX_VALUE);
             scoreDocs = topDocs.scoreDocs;
             cursor = OPENED;
+            
+            this.schema = dataReaderPredicate.getDataStore().getSchema();
 
-            this.queryTokens = Utils.tokenizeQuery(dataReaderPredicate.getLuceneAnalyzer(),dataReaderPredicate.getQueryString());
+            if (this.dataReaderPredicate.getIsSpanInformationAdded()) {

```

Comment:
DataReader still doesn't know whether it's Regex or not. The logic inside "if" is only useful for span information. So the if an operator(like regex) doesn't want span, the code will be skipped.